modules/keyless-authentication-robot-accounts.adoc

```diff
@@ -0,0 +1,268 @@
++
+[source,terminal]
+----
+$ curl -X POST "http://localhost:8080/realms/master/protocol/openid-connect/token" <1>
+-H "Content-Type: application/x-www-form-urlencoded" \
+-d "client_id=quaydev" <2>
+-d "client_secret=g8gPsBLxVrLo2PjmZkYBdKvcB9C7fmBz" <3>
+-d "grant_type=authorization_code"
+-d "code=ea5b76eb-47a5-4e5d-8f71-0892178250db.5c9bce22-6b85-4654-b716-e9bbb3e755bc.cdffafbc-20fb-42b9-b254-866017057f43" <4>
+----
+<1> The `protocol/openid-connect/token` endpoint found on the *Realm settings* page of the Red Hat SSO Keycloak UI.
+<2> The Client ID used for this procedure.
+<3> The Client Secret for the Client ID. 
+<4> The code returned from the redirect URI.
++
+.Example output
++
+[source,terminal]
+----
+{"access_token":"eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJTVmExVHZ6eDd2cHVmc1dkZmc1SHdua1ZDcVlOM01DN1N5T016R0QwVGhVIn0...",
+"expires_in":60,"refresh_expires_in":1800,"refresh_token":"eyJhbGciOiJIUzUxMiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJiNTBlZTVkMS05OTc1LTQwMzUtYjNkNy1lMWQ5ZTJmMjg0MTEifQ.oBDx6B3pUkXQO8m-M3hYE7v-w25ak6y70CQd5J8f5EuldhvTwpWrC1K7yOglvs09dQxtq8ont12rKIoCIi4WXw","token_type":"Bearer","not-before-policy":0,"session_state":"5c9bce22-6b85-4654-b716-e9bbb3e755bc","scope":"profile email"}
+----
+
+. Store the `access_token` from the previously step, as it will be exchanged for a {productname} robot account token in the following procedure.
+
+[id="setting-robot-federation"]
+== Setting up a robot account federation by using the {productname} v2 UI
+
+The following procedure shows you how to set up a robot account federation by using the {productname} v2 UI.
+
+.Prerequisites
+
+* You have created an organization. The following example uses `fed_test`.
+* You have created a robot account. The following example uses `fest_test+robot1`.
+* You have configured a OIDC for your {productname} deployment. The following example uses Red Hat SSO Keycloak.
+
+.Procedure
+
+. On the Keycloak main page:
+
+.. Select the appropriate realm that is authenticated for use with {productname}. Store the issuer URL, for example, `\https://keycloak-auth-realm.quayadmin.org/realms/quayrealm`.
+
+.. Click *Users* -> the name of the user to be linked with the robot account for authentication. You must use the same user account that you used when generating the OAuth2 access token.
+
+.. On the *Details* page, store the *ID* of the user, for example, `449e14f8-9eb5-4d59-a63e-b7a77c75f770`.
+
+. On your {productname} registry:
+
+.. Navigate to *Organizations* and click the name of your organization, for example, *fed_test*.
+
+.. Click *Robot Accounts*.
+
+.. Click the menu kebab -> *Set robot federation*.
+
+.. Click the *+* symbol. 
+
+.. In the popup window, include the following information:
++
+* *Issuer URL*: `\https://keycloak-auth-realm.quayadmin.org/realms/quayrealm`, or the URL of your Keycloak realm.
+* *Subject*: `449e14f8-9eb5-4d59-a63e-b7a77c75f770`, or the *ID* of your keycloak user. 
```

Comment:
Here need to use the subject, can't use the **ID** of the user.

Reply:
Sorry, I meant "which is the *ID* of your keycloak user. not one or the other. sorry for the confusion.